Zend/zend_object_handlers.c

```diff
@@ -917,6 +935,14 @@ static zval *zend_std_get_property_ptr_ptr(zval *object, zval *member, int type,
 	fprintf(stderr, "Ptr object #%d property: %s\n", Z_OBJ_HANDLE_P(object), ZSTR_VAL(name));
 #endif
 
+	if (UNEXPECTED(zobj->ce->ce_flags & ZEND_ACC_HAS_TYPE_HINTS)) {
+		zend_property_info *prop_info = zend_hash_find_ptr(&zobj->ce->properties_info, name);
+		
+		if (prop_info && prop_info->type) {
+			return NULL;
+		}
```

Comment:
This makes $obj->prop++; and and $obj->prop+=2; slower. They will require additional read/write calls.